src/SingleCellAnnotator.imjoy.html

```diff
@@ -434,27 +450,39 @@
           body: JSON.stringify(sendData)
         });
       const res= await response.json(); 
-      if(!res.success){
+      if(res.success){
+        this.current_image_data.annotation_status=true;
+      }
+      else{
         alert('Failed to upload the annotaiton, error: ' + res.error);
         throw new Error('Failed to upload the annotaiton.')
       };
     },
     async annotation_mode_changed(){
       if(this.annotation_mode){
+        this.current_image_data.mask_adjusted=false;
         await this.extendImage();
+        this.removeMaskAnnotation(); // remove the current mask if adjust mask
         this.startAnnotation(this.vector_source, this.freehand_mode)
       }
       else{
+        if(this.current_image_data.mask_adjusted==false){
```

Comment:
Replace to `if(!this.current_image_data.mask_adjusted){`

Reply:
changed to '==='.